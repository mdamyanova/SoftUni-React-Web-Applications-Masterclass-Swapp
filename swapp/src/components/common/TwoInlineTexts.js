import React, { useContext } from 'react';
import PropTypes from 'prop-types';
import { Box, Text } from 'rebass';
import ThemedHeading from './ThemedHeading';

const TwoInlineTexts = ({ firstText, secondText, align }) => {
  const textStyle = { display: 'inline-block', fontSize: 1 };

  return (
    <Box sx={{ textAlign: align }}>
      <Text
        css={`
          color: var(--default-font-color);
        `}
        px={2}
        {...textStyle}
      >
        {firstText}
      </Text>
      <ThemedHeading style={{ ...textStyle }} text={secondText} />
    </Box>
  );
};

TwoInlineTexts.propTypes = {
  firstText: PropTypes.string.isRequired,
  secondText: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  align: PropTypes.string
};

TwoInlineTexts.defaultProps = {
  align: 'left'
};

export default TwoInlineTexts;
