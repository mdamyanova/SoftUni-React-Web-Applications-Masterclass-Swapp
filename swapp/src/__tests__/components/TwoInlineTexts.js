import React from "react";
import { shallow, mount } from "enzyme";
import TwoInlineTexts from "../../components/common/forms/TwoInlineTexts";

describe("<TwoInlineTexts />", () => {
  it("should match snapshot", () => {
    expect(
      shallow(<TwoInlineTexts firstText="first" secondText="second" />)
    ).toMatchSnapshot();
  });

  it("should have props", () => {
    const wrapper = mount(
      <TwoInlineTexts firstText="first" secondText="second" />
    );
    expect(wrapper.props().firstText).toEqual("first");
    expect(wrapper.props().secondText).toEqual("second");
  });
});
