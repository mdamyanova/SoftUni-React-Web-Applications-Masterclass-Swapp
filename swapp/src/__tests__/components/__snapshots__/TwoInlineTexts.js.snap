// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TwoInlineTexts /> should match snapshot 1`] = `
<Styled(div)
  sx={
    Object {
      "textAlign": "left",
    }
  }
>
  <ForwardRef
    css="
          color: var(--default-font-color);
        "
    display="inline-block"
    fontSize={1}
    px={2}
  >
    first
  </ForwardRef>
  <ThemedHeading
    style={
      Object {
        "display": "inline-block",
        "fontSize": 1,
      }
    }
    text="second"
  />
</Styled(div)>
`;
